cpp/oneapi/dal/data/undirected_adjacency_array_graph.hpp

```diff
@@ -124,9 +124,6 @@ template <typename G>
 auto get_vertex_neighbors_impl(const G &g, const vertex_type<G> &vertex)
```

Comment:
why do we have impls in this header file?

Reply:
> why do we have impls in this header file?

As these are inner service functions for concrete graph type. We cannot left them only in cpp. This hpp us the most suitable.

Reply:
Are there any rules on having 'impl' functions in detail namespace?

Reply:
impl functions can be hidden backend if their decl is used in cpps only (not our case). In detail folder if we need.

Reply:
I still do not understand: we have the file with graph declaration and put implementation specifics here as well? If so, I'm as a user want to read only apis, not your implementation details